src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -542,8 +544,15 @@ public struct CompilationAnalysisContext
         /// </summary>
         public CancellationToken CancellationToken { get { return _cancellationToken; } }
 
+        /// <summary>
+        /// Whether or not calling <see cref="AddOutputFile(string, SourceText)"/> is allowed.
+        /// </summary>
+        [MemberNotNullWhen(true, nameof(_addOutputFile))]
+        public bool CanAddOutputFile => _addOutputFile != null;
+
+        [Obsolete("Analysis contexts should not be directly instantiated", error: false)]
```

Comment:
Why is this being obsoleted as part of this change? These APIs were made public as it was believed it would be useful for third-party analysis hosts. If that is no longer something we want to support, we should create a separate issue, get the design change approved from compiler team and then do it as part of a separate PR.

Reply:
i don't see how hosts can use these.  we have internal constructors that bypass these and pass in the actual data needed.  It looked like we were unintentionally making things public.  I can undo the Obsolete here, but it's super vlauable for us to at least make sure we're not calling the bad legacy constructors.

Reply:
note: this is the pattern elsewhere in this type.  we don't make constructors public and we oureslves call the internal constructors passing along all data.  This effectively indicates that the public constructor surface area here is effectively legacy.

Reply:
Seems like all changes in this file made to existing context structs can be reverted.

Reply:
yup